packages/ember-data/lib/serializers/rest-serializer.js

```diff
@@ -40,8 +40,12 @@ function coerceId(id) {
 
   ```js
   App.ApplicationSerializer = DS.RESTSerializer.extend({
-    keyForAttribute: function(attr) {
-      return Ember.String.underscore(attr).toUpperCase();
+    keyForAttribute: function(attr, method) {
+      if (method === 'serialize') {
+        return Ember.String.underscore(attr).toUpperCase();
+      } else {
+        return Ember.String.camelize(attr.toLowerCase());
+      }
```

Comment:
The `keyForAttribute` method's `attr` parameter is always the name of the attribute as defined on the Model. So if the API you are working with consistently uses the same keys when serializing and deserializing then you should not need to include a check on the method argument.

Using `method` method is usually reserved for the rare case where an API returns data using one key, but then expects it to be listed under a different key when updating.